File: python/test/canonical_test.py
# -*- coding: utf-8 -*-

import unittest
import re
import os
import sys

sys.path.append(os.path.join(os.path.dirname(__file__), "..", "lib"))
from ironnews_utility import IronnewsUtility

class TestCanonical(unittest.TestCase):
  def setUp(self):
    pass

  def test_get_canonical_url(self):
    file = open(os.path.join(os.path.dirname(__file__), "..", "..", "canonical_cases.txt"))
    for line in file:
      value, expected = re.split(r"\t", line.strip())
      self.assertEqual(expected, IronnewsUtility.get_canonical_url(value))
    file.close()

if __name__ == "__main__":
  unittest.main()

File: ruby/test/canonical_test.rb
#! ruby -Ku

$:.unshift(File.join(File.dirname(__FILE__), "..", "lib"))

require "test/unit"
require "ironnews_utility"

class CanonicalTest < Test::Unit::TestCase
  def setup
    @module = IronnewsUtility
  end

  def test_get_canonical_url
    cases = File.open(File.join(File.dirname(__FILE__), "..", "..", "canonical_cases.txt"), "rb") { |file|
      file.map { |line|
        line.chomp.split(/\t/)
      }
    }
    cases.each { |value, expected|
      assert_equal(expected, @module.get_canonical_url(value), value)
    }
  end
end
